### PR TITLE
lambda_expressions / it-academy.by / JC1-117-22 / Aliaksei Kharko / Done tasks and tests 

### PR DESCRIPTION
**Pull request by KharkoAO**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/Task01.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+public class Task01 {
+    public Map<String, String> convertArrayInMap(String[] array) {
+        return Stream.iterate(0, a -> a < array.length, a -> a + 2)
+                .collect(Collectors.toMap(a -> array[a], a -> array[a + 1]));
+    }
+
+    public String[] convertMapInArray(Map<String, String> map) {
+        return map.entrySet().stream()
+                .flatMap(m -> Stream.of(m.getKey(), m.getValue())).toArray(String[]::new);
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/Task01.java:15]:
```suggestion
                .flatMap(m -> Stream.of(m.getKey(), m.getValue()))
                .toArray(String[]::new);
```

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/Task01Demo.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    public static final Logger logger = Logger.getLogger(Task01.class.getName());
+
+    public static void main(String[] args) {
+        Task01 task01 = new Task01();
+        String[] array = {"-i", "in.txt", "--limit", "40", "-d", "1", "-o", "out.txt"};
+        Map<String, String> map = Map.of("-i", "in.txt", "--limit", "40", "-d", "1", "-o", "out.txt");
+
+        logger.info(task01.convertArrayInMap(array).toString());
+        logger.info(Arrays.toString(task01.convertMapInArray(map)));
+    }
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/Task01Demo.java:17]: please write tests

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.rakovets.course.java.core.practice.lambda_expressions.maths_statistics;
+
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class MathsStatistics {
+    public List<Integer> creationSetRandomNumbers() {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java:8]: bad API

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        return new Random().ints(10, 0, 11)
+                .boxed()
+                .collect(Collectors.toList());
+    }
+
+    public long countingAmountEvenNumbers(List<Integer> listNumbers) {
+        return listNumbers.stream()
+                .filter(x -> x % 2 == 0).count();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java:16]:
```suggestion
                .filter(x -> x % 2 == 0)
                .count();
```

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public long countingAmountOddNumbers(List<Integer> listNumbers) {
+        return listNumbers.stream()
+                .filter(x -> x % 2 != 0).count();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java:21]: alike

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public long countingAmountNumbersEqualZero(List<Integer> listNumbers) {
+        return listNumbers.stream()
+                .filter(x -> x == 0).count();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java:26]: alike

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public long countingAmountGivenNumbers(List<Integer> list, int number) {
+        return list.stream()
+                .filter(x -> x == number).count();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java:31]: alike

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/tv/InformationSystemTv.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class InformationSystemTv {
+    public List<Television> televisions;
+
+    public InformationSystemTv(List<Television> televisions) {
+        this.televisions = televisions;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/tv/InformationSystemTv.java:12]: Where do you use this field?

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/city_helter/CityHelper.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.stream.Collectors;
+
+public class CityHelper {
+    public List<String> getUniqueCityNames(List<String> listCities) {
+        return listCities.stream()
+                .distinct()
+                .collect(Collectors.toList());
+    }
+
+    public List<String> getCityNamesGivenLength(List<String> listCities) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/city_helter/CityHelper.java:13]: bad API

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public class CityHelperTest {
+    CityHelper city = new CityHelper();
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelperTest.java:13]: use `@BeforeAll`

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatisticDemo.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+        logger.info("Random numbers: " + listNumber);
+        logger.info("Even numbers: " + statistics.countingAmountEvenNumbers(listNumber)
+                + " in: " + listNumber);
+        logger.info("Odd numbers: " + statistics.countingAmountOddNumbers(listNumber)
+                + " in: " + listNumber);
+        logger.info("Amount 0: " + statistics.countingAmountNumbersEqualZero(listNumber)
+                + " in: " + listNumber);
+        logger.info("Amount " + number + " : " + statistics.countingAmountGivenNumbers(listNumber, number)
+                + " in: " + listNumber);
+    }
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatisticDemo.java:25]: please, write tests

**Review by rakovets — CHANGES_REQUESTED** [review]
need to fix

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/Task01.java (hunk 1 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/Task01.java]: This fixed

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/Task01Demo.java (hunk 2 above)
Comment (KharkoAO) [reply src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/Task01Demo.java]: Done

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java (hunk 3 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java]: This fixed

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java (hunk 4 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java]: This fixed

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java (hunk 5 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java]: This fixed

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java (hunk 6 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java]: This fixed

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java (hunk 7 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/maths_statistics/MathsStatistics.java]: This fixed

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/tv/InformationSystemTv.java (hunk 8 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/tv/InformationSystemTv.java]: This fixed

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/city_helter/CityHelper.java (hunk 9 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/city_helter/CityHelper.java]: This fixed

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatisticDemo.java (hunk 11 above)
Comment (KharkoAO) [reply src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatisticDemo.java]: Done

**Pull request closed without merging**